src/buffer.jl

```diff
@@ -26,17 +26,18 @@ mutable struct Buffer
     # the total number of transcoded bytes
     transcoded::Int64
 
-    function Buffer(size::Integer)
-        return new(Vector{UInt8}(undef, size), 0, 1, 1, 0)
+    function Buffer(data::Vector{UInt8}, keepbytes::Integer=length(data))
```

Comment:
I rather call `keepbytes` as `marginpos` so we don't introduce yet another term that we have to document. Alternatively, we should document what `keepbytes` is.